config/routes.rb

```diff
@@ -1,3 +1,19 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root to: 'pages#home'
+
+
+    resourses :sponsors
+    resources :events do
+                          companies
+                          workshops do users
+                          end
+                          end
+    "event/book", to: "event#book"
+    "event/sponsor" to: "sponsor#sponsor"
+    "event/workshop/attend" to: "workshop/attend"
```

Comment:
I think you attend an event, you're routing to a nested `Workshop::Attend` controller, that you don't have